vpd-tool/src/vpd_tool.cpp

```diff
@@ -0,0 +1,59 @@
+#include "vpd_tool.hpp"
+
+#include "constants.hpp"
+#include "types.hpp"
+#include "utils.hpp"
+
+#include <iostream>
+
+namespace vpd
+{
+void VpdTool::readKeyword(const std::string& i_fruPath,
+                          const std::string& i_recordName,
+                          const std::string& i_keywordName, bool i_onHardware,
+                          const std::string& i_fileToSave)
+{
+    types::DbusVariantType l_keywordValue;
+    if (i_onHardware)
+    {
+        std::cout << "Read from h/w" << std::endl;
+        // TODO: uncomment when implementation of the API goes in
+        /*l_keywordValue = utils::readKeywordFromHardware(
+            constants::vpdManagerService,
+           constants::vpdManagerObjectPath,
+            constants::vpdManagerIntfName, i_fruPath,
+            make_tuple(i_recordName, i_keywordName));*/
+    }
+    else
+    {
+        std::string l_fullObjectPath(constants::baseInventoryPath + i_fruPath);
+
+        l_keywordValue = utils::readDbusProperty(
+            constants::inventoryManagerService, l_fullObjectPath,
+            constants::ipzVpdIntf + i_recordName, i_keywordName);
+    }
+
+    if (const auto l_value = std::get_if<types::BinaryVector>(&l_keywordValue))
+    {
+        const std::string& l_keywordStrValue =
+            utils::getPrintableValue(*l_value);
+
+        // TODO: handle in case result should be redirect to a file.
+        (void)i_fileToSave;
+
+        nlohmann::json l_resultInJson = nlohmann::json::object({});
+        nlohmann::json l_keywordValInJson = nlohmann::json::object({});
+
+        l_keywordValInJson.emplace(i_keywordName, l_keywordStrValue);
```

Comment:
can we add the values inline while constructing the json object?
in both line 47 and 48